// Copyright (c) 2025 The Brave Authors. All rights reserved.
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this file,
// You can obtain one at https://mozilla.org/MPL/2.0/.

syntax = "proto3";

package ai_chat.store;

option optimize_for = LITE_RUNTIME;

//
// Structured data stored by the client, usually in SQL
//

message ImageContentBlockProto {
  string image_url = 1;
}

message TextContentBlockProto {
  string text = 1;
}

message ContentBlockProto {
  oneof content {
    ImageContentBlockProto image_content_block = 1;
    TextContentBlockProto text_content_block = 2;
  }
}

message ToolUseEventProto {
  string tool_name = 1;
  string id = 2;

  // We can't use google.protobuf.Struct in Chromium
  // so we use JSON string for something that's easily convertable
  // to DictionaryValue.
  string arguments_json = 3;
  repeated ContentBlockProto output = 4;

  // Security metadata from server indicating if tool call is allowed.
  // When not present, no security check was performed by the server.
  optional bool security_metadata_allowed = 5;

  // Reasoning provided by server when tool call is not allowed.
  // Only meaningful when security_metadata_allowed is false.
  optional string security_metadata_reasoning = 6;
}

message WebSourceProto {
  string title = 1;
  string url = 2;
  string favicon_url = 3;
}

message WebSourcesEventProto {
  repeated WebSourceProto sources = 1;
}

message SmartModeEntryProto {
  string shortcut = 1;
  string prompt = 2;
}
